src/Rules/MissingTypehintCheck.php

```diff
@@ -133,4 +139,27 @@ public function getNonGenericObjectTypesWithGenericClass(Type $type): array
 		return $objectTypes;
 	}
 
+	/**
+	 * @param \PHPStan\Type\Type $type
+	 * @return \PHPStan\Type\Type[]
+	 */
+	public function getCallablesWithMissingPrototype(Type $type): array
+	{
+		if (!$this->checkMissingCallablePrototype) {
+			return [];
+		}
+
+		$result = [];
+		TypeTraverser::map($type, static function (Type $type, callable $traverse) use (&$result): Type {
+			if (
+				($type instanceof CallableType && $type->isCommonCallable()) ||
+				($type instanceof ObjectType && $type->getClassName() === \Closure::class)) {
```

Comment:
This is the only thing I'm not happy with. I think there should be a better, more generalized way to do this, but if there is one it's not apparent to me.

Reply:
This is probably fine... There's `Type::isCallable()` and `Type::getCallableParametersAcceptors()` but there's no way to extract whether it's general callable or already specified callable from that.

So as long as this place is synchronized with `TypeNodeResolver::resolveCallableTypeNode()`, we're fine.